store/src/main/java/org/apache/rocketmq/store/ha/HAService.java

```diff
@@ -452,11 +452,13 @@ private boolean dispatchReadRequest() {
                     }
 
                     if (diff >= (msgHeaderSize + bodySize)) {
-                        byte[] bodyData = new byte[bodySize];
-                        this.byteBufferRead.position(this.dispatchPosition + msgHeaderSize);
-                        this.byteBufferRead.get(bodyData);
+                        if (bodySize > 0) {
+                            byte[] bodyData = new byte[bodySize];
+                            this.byteBufferRead.position(this.dispatchPosition + msgHeaderSize);
```

Comment:
Here will cause an unnecessary copy of the data.

If boySize == 0, just ignore the appendToCommitlog.